tests_integration/test_init.py

```diff
@@ -0,0 +1,106 @@
+    url = "https://github.com/snowflakedb/snowflake-cli/this-url-does-not-exist/"
+    path = "a_new_project"
+    result = runner.invoke(["init", path, "--template-source", url])
+    assert result.exit_code == 1
+    assert "Error" in result.output
+    assert url in result.output
+    assert "not exist" in result.output
+    assert not (temporary_working_directory / path).exists()
+
+
+@pytest.mark.integration
+def test_not_existing_template(runner, temporary_working_directory):
+    url = "https://github.com/snowflakedb/snowflake-cli/"
+    template = "this/template/does/not/exist"
+    path = "a_new_project"
+    result = runner.invoke(
+        ["init", path, "--template-source", url, "--template", template]
+    )
+    assert result.exit_code == 1
+    assert "Error" in result.output
+    assert f"Template '{template}' cannot be found under" in result.output
+    assert url in result.output
+
+
+@pytest.mark.integration
+def test_missing_template_yml(runner, temporary_working_directory):
+    url = "https://github.com/snowflakedb/snowflake-cli/"
+    path = "a_new_project"
+    result = runner.invoke(["init", path, "--template-source", url])
+    assert result.exit_code == 1
+    assert "Error" in result.output
+    assert f"Template does not have template.yml file." in result.output
+
+
+@pytest.mark.integration
+def test_template_contents(runner, temporary_working_directory, test_root_path):
+    url = "https://github.com/snowflakedb/snowflake-cli/"
+    template = "tests/test_data/projects/project_templating"
+    remote_path = "from_remote"
+
+    # remote template
+    result = runner.invoke(
+        [
+            "init",
+            remote_path,
+            "--template-source",
+            url,
+            "--template",
+            template,
+            "-D required_project_name=a new project",
+            "--no-interactive",
+        ]
+    )
+    assert result.exit_code == 0, result.output
+
+    # local template
+    local_template = test_root_path.parent / template
+    local_path = "from_local"
+    result = runner.invoke(
+        [
+            "init",
+            local_path,
+            "--template-source",
+            local_template,
+            "-D required_project_name=a new project",
+            "--no-interactive",
+        ]
+    )
+    assert result.exit_code == 0, result.output
+
+    # compare results
+    def _list_contents(root: Path) -> Set[Path]:
+        return set(x.relative_to(root) for x in root.rglob("*"))
+
+    assert _list_contents(Path(local_path)) == _list_contents(Path(local_path))
+
+
+@pytest.mark.integration
+@pytest.mark.parametrize("template", ["example_snowpark", "example_streamlit"])
+def test_default_url(runner, temporary_working_directory, template):
+    path = "a_new_project"
+    runner = runner.invoke(["init", path, "--template", template, "--no-interactive"])
+    assert runner.exit_code == 0, runner.output
+    assert (Path(path) / "snowflake.yml").exists()
```

Comment:
Is it working for many cases? I think if firsts case pass, the rest of them will pass too.

Reply:
Removed multiple cases